crates/bevy_ecs/src/world/mod.rs

```diff
@@ -237,7 +238,13 @@ impl World {
     /// ```
     #[inline]
     pub fn get_entity(&self, entity: Entity) -> Option<EntityRef> {
-        let location = self.entities.get(entity)?;
+        let location = match self.entities.get(entity) {
+            Ok(entity_location) => entity_location,
+            Err(err) => {
+                warn!("Error getting entity {:?}: {:?}", entity, err,);
```

Comment:
I think `warn!` is a too strong log level. The user may very well just not care about if the entity exists or not. I think it would be better to return `Result<EntityRef, GetEntityError>` where `GetEntityError` has a `Debug` implementation that shows all this information.

Reply:
I agree. This method feels too "low level" and could be called in too many different situations to call this case a warning / make it unrecoverable. I'd prefer to preserve the simplicity of the `Option` return type (like `std`'s collection types) if possible, but if we really need to expose multiple "failure" states, it probably makes sense to just return an error here.

Reply:
I... can't remember why I wanted a warn here :(
I could weakly argue that the user should not ignore this, but the `Option` already achieve that. 

For the error reason, I think knowing it can help the user debug issues looking at the log but is not very useful from the code. Should I change the log to info or debug?

Reply:
I honestly don't want to log at all here. If some subset of users need to care about the various reasons an entity might not exist, then imo we need to return a Result here.